[ews] Reduce excessive logging due to github webhooks

aj062 opened the pull request:
#### 25e7f0c5753514ce60c760ce326e0d170aee1ae6
<pre>
[ews] Reduce excessive logging due to github webhooks
<a href="https://bugs.webkit.org/show_bug.cgi?id=253529">https://bugs.webkit.org/show_bug.cgi?id=253529</a>

Reviewed by Jonathan Bedard.

Reduce excessive logging due to github webhooks. This huge logging makes twistd.log unreadable and isn&apos;t really useful to use.

* Tools/CISupport/ews-build/events.py:
(logging_disabled): context manager to disable the log.msg function appropriately.
(logging_disabled.__init__):
(logging_disabled.log_disabled):
(logging_disabled.__enter__):
(logging_disabled.__exit__):
(GitHubEventHandlerNoEdits._get_payload):
(GitHubEventHandlerNoEdits.handle_pull_request): Updated logs to include hash (to help in debugging ews history of a hash).

Canonical link: <a href="https://commits.webkit.org/261373@main">https://commits.webkit.org/261373@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/100914a34c5f417eaecfae4faa04715acffc079b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/111373 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/20511 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/87/builds/3069 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/WinCairo-EWS "Waiting in queue, processing has not started yet") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/11/builds/115327 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/21878 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/11604 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/87/builds/3069 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/19/builds/117137 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/21878 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/104041 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/21878 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [⏳ 🧪 mac-wk1 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/104041 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/13012 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/104041 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/13522 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/11604 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [  ~~🧪 services~~](https://ews-build.webkit.org/#/builders/20/builds/110146 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/18965 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/15481 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4319 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/CISupport/ews-build/events.py:
```diff
@@ -263,6 +263,21 @@ def stopService(self):
         self._stepFinishedConsumer.stopConsuming()
 
 
+class logging_disabled(object):
```
Might consider `@contextlib.contextmanager` here, it would let you make this a function instead of a class.

aj062 replied:
Interesting. Although this particular change is probably not worth spending additional time.

JonWBedard reviewed (approved):
I'm happy with this change as is, just wanted to offer a possibility for what I think is code simplification. I think `contextlib.contextmanager` is more a mater of taste than anything, though, so definitely doesn't block landing.

aj062 committed:
[ews] Reduce excessive logging due to github webhooks
https://bugs.webkit.org/show_bug.cgi?id=253529

Reviewed by Jonathan Bedard.

Reduce excessive logging due to github webhooks. This huge logging makes twistd.log unreadable and isn't really useful to use.

* Tools/CISupport/ews-build/events.py:
(logging_disabled): context manager to disable the log.msg function appropriately.
(logging_disabled.__init__):
(logging_disabled.log_disabled):
(logging_disabled.__enter__):
(logging_disabled.__exit__):
(GitHubEventHandlerNoEdits._get_payload):
(GitHubEventHandlerNoEdits.handle_pull_request): Updated logs to include hash (to help in debugging ews history of a hash).

Canonical link: https://commits.webkit.org/261373@main
